domain/schema/model/sql/0022-resource.sql

```diff
@@ -133,3 +134,14 @@ CREATE TABLE unit_resource (
     REFERENCES unit (uuid),
     PRIMARY KEY (resource_uuid, unit_uuid)
 );
+
+CREATE TABLE resource_oci_image_metadata_store (
+    resource_uuid TEXT NOT NULL,
+    registry_path TEXT NOT NULL,
+    username TEXT,
+    password_hash TEXT,
+    password_salt TEXT,
```

Comment:
Is hash the right term here? Hash implies a one-way hash, when we need to encrypt the password (we need the value back out). The nomenclature is "box" for the value, so `password_hash` would become `password_box`, which needs to be base64 encoded. If we're using a secretbox here, we should use `nonce` and we could hardcode the key in the table.

I'm reluctant to store the nonce on the same table as well, I think the nonce should be something else as well.

Essentially you need three values:

```sql
...
password_box TEXT,
password_key TEXT,
...
```

And the nonce is some value that's not obvious.

Reply:
Didn't see this comment in time. After conversation, #18351, will remove the username and password bits from the table until a secure way to handle the data has been chosen.